weldx/util/util.py

```diff
@@ -291,28 +299,18 @@ def compare_nested(
             When a or b is not a nested structure.
 
         """
-
-        def _enter_replace_sets(p, k, v):
-            if isinstance(v, set):
-                return [], ((i, j) for i, j in enumerate(sorted(list(v))))
-            return iterutils.default_enter(p, k, v)
-
-        try:
-            a = iterutils.remap(a, enter=_enter_replace_sets)
-            b = iterutils.remap(b, enter=_enter_replace_sets)
-        except TypeError:
-            raise TypeError("either a or b are not a nested data structure.")
-
         # we bind the input structures a, b to the visit function.
         visit = functools.partial(_EqCompareNested._visit, a=a, b=b)
 
         try:
-            iterutils.remap(a, visit=visit, reraise_visit=True)
+            iterutils.remap(
+                a, visit=visit, reraise_visit=True, enter=_EqCompareNested._enter
+            )
         # Key not found in b, values not equal, more elements in a than in b
         except (KeyError, RuntimeError, IndexError):
             return False
         except TypeError:
-            raise TypeError("either a or b are not a nested data structure.")
+            raise TypeError("One of a or b is not a nested data structure (or a set).")
```

Comment:
If we use sets as `a` and `b`, shouldn't the function just return `a==b` instead of raising an exception?

Reply:
it would be nice but the way it is implemented using `remap` now `set` is not considered a nested structure (by the `enter` function) so you cannot apply it here

In the end I guess we have basic functionality for sets that appear in nested structures which hopefully covers most of our use cases